network/test/meshengine_test.go

```diff
@@ -45,7 +45,7 @@ func (suite *MeshEngineTestSuite) SetupTest() {
 	const count = 10
 	logger := zerolog.New(os.Stderr).Level(zerolog.ErrorLevel)
 	log.SetAllLoggers(log.LevelError)
-	suite.ids, _, suite.nets = GenerateIDsMiddlewaresNetworks(suite.T(), count, logger, 100, nil, !DryRun)
+	suite.ids, _, suite.nets = GenerateIDsMiddlewaresNetworks(suite.T(), count, logger, 100, nil, !DryRun, unittest.WithAllRoles())
```

Comment:
will the node count still be 10? (with the `WithAllRoles` option)

Reply:
`WithAllRoles` just (semi-uniformly) distributes the roles among the nodes. It does not decide on the number of the nodes. To construct the topology healthily, we need at least two nodes from each role (2 * 5 roles = 10).